### PR TITLE
Refactor/planetarium

### LINKED ISSUE #31: Polish Planetarium
Planets are displaying and sorting based on defined criteria.

- [x] When a button is pressed to sort, planet information must reflect this change(ie. when sorted by length of day each planet will show that number, followed by `hours`). Additional flavor text can be added to dev content.

- [x] Test out changing text color based on planet? So if planet icon is red or yellow, text is light yellow found in docs, for blue planets it would be light blue. If this is ugly don't continue.

### PR DESCRIPTION
**Pull request by ConnorAndersonLarson**
# PR PlanetParty
**Connor A-L, Alex T & Katie B**

### What does this do?

- [ ] Fix
- [ ] Feature
- [X] Refactor
- [X] Style
- [ ] Testing

### Are there any known bugs?

- [X] No
- [ ] Yes (if so please disclose in Notes for Reviewer & Next Iterations)

### Summary of the changes
Planet data on main page now changes based on sort parameters, with descriptor (e.g. sorting by moons will return the planets with data displayed as 'x moons' based on how many moons each planet has).

When a planet is hovered over the text changes color to either a yellow or blue, dependent on the color of the planet (the exception being mercury, which has no color, currently the text remains gray)

### What ticket(s) do the changes resolve?
Close #31 

Please Link Issues to this PR as necessary.
#31 

### Notes for the Reviewer
This has only been tested on desktop view. May need to tweak display for touch based devices.

### How to Test Changes?
Enter this branch, run `npm start` 
Use sort function to change planet info/rankings, observe data below planet name
Hover over planets to view text change color; this is on a one second transition.

### Next Iterations
N/A

### CODE REVIEW

File: src/components/Planetarium/Planetarium.tsx [hunk 1]
```diff
@@ -3,15 +3,34 @@ import { PlanetBio, AllData } from '../../interface';
 import './planetarium.css';
 import Planet from '../Planet/Planet';
 
-const Planetarium: React.FC<AllData> = ({ allPlanets }): JSX.Element => {
+const Planetarium: React.FC<AllData> = ({ allPlanets, sortKey }): JSX.Element => {
+
+  const addDescriptor = (data: any, key: any) => {
```
Comment (alexthompson207) [inline src/components/Planetarium/Planetarium.tsx]: could key be type string here? Does it need to be `key: any` and `data: any` ?

File: src/interface.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export interface PlanetBio {
+  data?: any;
```
Comment (alexthompson207) [inline src/interface.ts]: is data here a `string` ?

**Review by alexthompson207 — COMMENTED** [review]
This looks great, good work with adding planet facts dynamically to the Planet cards!

I'm a little confused what `data` and `key` are in the `addDescriptor` method, and wondering if they could have a type other than `any`?

File: src/interface.ts (hunk 2 above)
Comment (ConnorAndersonLarson) [reply src/interface.ts]: fixed!

File: src/components/Planetarium/Planetarium.tsx (hunk 1 above)
Comment (ConnorAndersonLarson) [reply src/components/Planetarium/Planetarium.tsx]: fixed! Just trying to figure out the initial solution and forgot to double back.

File: src/components/Planet/planet.css [hunk 3]
```diff
@@ -1,3 +1,7 @@
+.planet-card-name, .planet-card-fact{
+  transition: color 1s
```
Comment (knees4bees) [inline src/components/Planet/planet.css:2]: Ooh, love this!

File: src/components/Planet/planet.css [hunk 4]
```diff
@@ -9,8 +13,18 @@
...
+.planet-icon:hover ~ .Uranus {
+  color: #68CCD7;
+}
+
+.planet-icon:hover ~ .Venus,
+.planet-icon:hover ~ .Mars,
+.planet-icon:hover ~ .Jupiter,
+.planet-icon:hover ~ .Saturn {
+  color: #F5AE45;
+}
```
Comment (knees4bees) [inline src/components/Planet/planet.css:31]: I like that you matched the colors to the planets

File: src/components/Planetarium/Planetarium.tsx [hunk 5]
```diff
@@ -3,15 +3,34 @@ import { PlanetBio, AllData } from '../../interface';
...
+    } else if ( key === 'mass') {
+      descriptor = 'kilograms';
+    } else if ( key === 'length_of_day') {
+      descriptor = 'hours';
+    } else if ( key === 'length_of_year') {
+      descriptor = 'days'
+    } else if ( key === 'number_of_moons') {
+      descriptor = 'moons'
+    } else if ( key === 'gravity') {
+      descriptor = 'metres per second squared'
```
Comment (knees4bees) [inline src/components/Planetarium/Planetarium.tsx:21]: Minorest of minor nitpicks here but let's make it "kilometers" and "meters" or "kilometres" and "metres" just to be consistent

File: src/components/Planetarium/Planetarium.tsx [hunk 6]
```diff
@@ -3,15 +3,34 @@ import { PlanetBio, AllData } from '../../interface';
...
+    }
+    return (`${data} ${descriptor}`)
```
Comment (knees4bees) [inline src/components/Planetarium/Planetarium.tsx:23]: Great job getting this to work!

File: src/components/SortBox/SortBox.css [hunk 7]
```diff
@@ -1,7 +1,4 @@
-* {
-  color: white
-}
```
Comment (knees4bees) [inline src/components/SortBox/SortBox.css]: Good call deleting this chunk. I had put it in temporarily just to see if I could get the text to show up and then forgot to remove it.

**Review by knees4bees — APPROVED** [review]
Solid work, @ConnorAndersonLarson!

**Comment by alexthompson207** [comment]
Great job with this!

**Pull request merged**